### ISSUE TITLE
DoubleEndedIterator

### ISSUE THREAD

**Issue by MggMuggins** [HarrisonMc555/array2d#1]
Hi, I've been enjoying your library in a couple of projects and I came upon a situation where I needed to reverse the columns of the matrix but not the rows.
```rust
for col in board.as_rows().iter().rev() {
    for elem in col.iter() {
        print!("{} ", elem);
    }
    println!();
}
```
That works, but I think it would probably be nice to have the iterators that are returned by `rows_iter` and `columns_iter` be double-ended, that way that code can be:
```rust
for col in board.rows_iter().rev() {
    for elem in col {
        print!("{} ", elem);
    }
    println!();
}
```
I think that'd probably be more efficient and since the length of the array is known at the time the iterator is created, should probably work. To be clear though, I haven't looked at the implementation so that may very well not be the case.

**Comment by HarrisonMc555** [comment]
Hello, @MggMuggins! Thanks for opening an issue—I'm very sorry it's taken me so long to get back to you. I assumed that I would have gotten some sort of email or notification when issues are created but I either didn't get one or missed it. Whoops!

I think that's a great idea! I'll look into it a little more and get back to you on how feasible it would be. I don't think it should be too bad.

Glad to hear you're enjoying it! If you don't mind me asking, what were you using it for? I'm just curious to see what use cases others have.

**Comment by MggMuggins** [comment]
This was just for a pet project to implement a basic chess UI. I just needed a low-cost way to store a 2d array that could have a variable size (wanted to implement more types of chess games), and ofc `Vec<Vec<_>>` is a bad pattern.

I think I ended up turning this into an ugly chain of iterator calls that I can no longer read, and because I needed some extra functionality related to rendering in the terminal I didn't end up needing this feature. Still a good idea to pursue it for the sake of your library though.

**Comment by HarrisonMc555** [comment]
Cool! Thanks for the info 😄 Board games were one area I explicitly thought this would work well in.

I just added a pull request (#2). If you have a second, do you want to look and see if you think it would have provided the functionality you were looking for? I'm guessing just looking at the tests I added should tell you one way or the other.

**Comment by HarrisonMc555** [comment]
Closed by #2
